rfcs/0000-upgradeable-lifecycle.md

```diff
@@ -0,0 +1,91 @@
+**Problem:**
+Currently, the lifecycle version used in kpack is not configurable, nor is it user upgradeable. Users must wait for a new kpack release to be able to upgrade the version of the lifecycle used by their builders.
+
+**Outcome:**
+We would like to decouple the lifecycle upgrade process from the kpack upgrade process, allowing users to upgrade the lifecycle on their own.
+
+**Actions to take:**
+A simple, monolithic approach that would keep a single lifecycle version for the whole cluster like it is today, but with the functionality fo configuring that version exposed. The main benefit of doing something like this it will likely be the least intrusive on user workflow. The lifecycle would continue to be something that is more back of mind for users. We would ship a version of the lifecycle with each kpack release like we do today. Additionally, since kpack has to be compatible with the platform api of the lifecycle, keeping the lifecycle version tied to kpack instead of a resource on a builder seems to make more sense right now. We would implement this by adding a few features to kp:
+
+1. Create a command in `kp` that would give users the ability to manually update their lifecycle:
+
+	`kp lifecycle upgrade --path "https://github.com/buildpacks/lifecycle/releases/download/v0.9.2/lifecycle-v0.9.2+linux.x86-64.tgz"`  
+	or  
+	`kp lifecycle upgrade --image "gcr.io/my-registry/lifecycle:v0.9.2"`  
+
+2. Add the concept of lifecycle version to our descriptor files so that `kp import` can upgrade the lifecycle. 
+
+One issue with this approach is that currently, the lifecycle image is passed to the kpack controller as an environment variable, so any change to the lifecycle image would require a kpack controller restart to propogate that change. We could get around this by introducing a `kpack-config` config map that would be monitored by the kpack controller for changes. To start, this config will probably just have the lifecycle version in it, but it does open the door for us to add other config options further down the road. An example `kpack-config` could look like:
```

Comment:
should we just use the existing `kp-config` config map used for the canonical values? having a `kp-config` and a `kpack-config` seems confusing to me